sotodlib/io/hk_utils.py

```diff
@@ -206,4 +210,78 @@ def get_hkaman(start, stop, config):
     data = get_grouped_hkdata(start, stop, config)
     hk_amans = make_hkaman(data)
     return hk_amans
+
+
+def detcosample_hkamans(hkconfig, detconfig, obs_id):
+    """
+    Given a detector observation, output an HK axismanager that cosampled
+    with detector timestreams.
+
+    Parameters:
+        hkconfig (str): The.yaml filename for loading HK fields and aliases.
+        detconfig (str): The .yaml config file matching readout id and detector ids.
+        obs_id (str): The observation id for a detector UFM. Helps determine
+                      the detector start and stop times needed for interpolation.
+
+    Notes
+    -----
+
+    An example detconfig file looks like:
+
+        data_prefix : "/mnt/so1/data/chicago-latrt"
+        hwp_prefix : "/mnt/so1/shared/site-pipeline/data_pkg/chicago-latrt/hwp/"
+        g3tsmurf_db: "/mnt/so1/shared/site-pipeline/data_pkg/chicago-latrt/g3tsmurf.db"
+    """
+    # load detector timestamps
+    SMURF = ls.G3tSmurf.from_configs(detconfig)
+    session = SMURF.Session()
+    obs = session.query(Observations).filter(Observations.obs_id == obs_id).one()
+    det_aman = ls.load_file([f.name for f in obs.files], archive=SMURF)
```

Comment:
This be much more useful if you allow the user to load their own AxisManager first, there are way too many ways people want to customize what data they have loaded for you to do it here. I think you just need to take in `aman` and an hkconfig. You can use `aman.timestamps[0]` and `aman.timestamps[-1]` to get the time ranges.